hack/build-binaries.sh

```diff
@@ -2,16 +2,24 @@
 
 set -e -x -u
 
-./hack/build.sh
+function get_latest_git_tag {
+  git describe --tags | grep -Eo '[0-9]+\.[0-9]+\.[0-9]+'
+}
+
+VERSION="${1:-`get_latest_git_tag`}"
+
+go fmt ./cmd/... ./pkg/... ./test/...
+go mod vendor
+go mod tidy
 
 # makes builds reproducible
 export CGO_ENABLED=0
-repro_flags="-ldflags=-buildid= -trimpath"
+LDFLAGS="-X github.com/k14s/kapp/pkg/kapp/version.Version=$VERSION -buildid="
 
-GOOS=darwin GOARCH=amd64 go build $repro_flags -o kapp-darwin-amd64 ./cmd/kapp/...
-GOOS=darwin GOARCH=arm64 go build $repro_flags -o kapp-darwin-arm64 ./cmd/kapp/...
-GOOS=linux GOARCH=amd64 go build $repro_flags -o kapp-linux-amd64 ./cmd/kapp/...
-GOOS=linux GOARCH=arm64 go build $repro_flags -o kapp-linux-arm64 ./cmd/kapp/...
-GOOS=windows GOARCH=amd64 go build $repro_flags -o kapp-windows-amd64.exe ./cmd/kapp/...
+GOOS=darwin GOARCH=amd64 go build -ldflags="$LDFLAGS" -trimpath -o kapp-darwin-amd64 ./cmd/kapp/...
+GOOS=darwin GOARCH=arm64 go build -ldflags="$LDFLAGS" -trimpath -o kapp-darwin-arm64 ./cmd/kapp/...
+GOOS=linux GOARCH=amd64 go build -ldflags="$LDFLAGS" -trimpath -o kapp-linux-amd64 ./cmd/kapp/...
+GOOS=linux GOARCH=arm64 go build -ldflags="$LDFLAGS" -trimpath -o kapp-linux-arm64 ./cmd/kapp/...
+GOOS=windows GOARCH=amd64 go build -ldflags="$LDFLAGS" -trimpath -o kapp-windows-amd64.exe ./cmd/kapp/...
 
-shasum -a 256 ./kapp-*
```

Comment:
We probably wanna keep this? Seems to have the same effect

Reply:
i'm not sure if the order is guaranteed. 

The order needs to match https://github.com/vmware-tanzu/carvel-kapp/pull/226/files/80ab962ddf17464474bed5b84d286a501e14dbc6#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R79-R83

Reply:
Makes sense to keep it this way and avoid the overhead!